File: Logic_HandsOn/C++/Problems on Digits,String,Array,Pointers etc/program27.cpp
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		a program which accept range from user and display all even numbers in between that range.
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////

#include<iostream>
using namespace std;

class Range
{
	private:
		int iStart;
		int iEnd;
		int iCnt;

	public:	

		Range(int iStart=0,int iEnd=0,int iCnt=0)
		{
		
			this->iStart=iStart;
			this->iEnd=iEnd;
		}

		void RangeDisplayEven()
		{
        		if (iStart>iEnd)
        		{
                		printf("Invalid Input");
        		}
		        else
        		{
		                for(iCnt=iStart;iCnt<=iEnd;iCnt++)
                		{
					if(iCnt%2==0)
					{
                        			cout<<iCnt<<"\n";
					}	
                		}
        		}
        		cout<<"\n";
		}
};		

int main()
{
        int iValue1=0,iValue2=0;

        cout<<"Enter starting point ";
        cin>>iValue1;
        
	cout<<"Enter ending point ";
        cin>>iValue2;

	Range *rObj = new Range(iValue1,iValue2);
        rObj->RangeDisplayEven();

        return 0;
}

File: Logic_HandsOn/Java/Problems on Array,Pattern etc/Program27.java
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		a program which accept range from user and display all even numbers in between that range.
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////


import java.util.*;

class Solution
{
	public int iStart,iEnd;

	public Solution(int iStart,int iEnd)
	{
		this.iStart = iStart;
		this.iEnd = iEnd;
	}

	public void RangeEvenDisplay()
	{
                        
		for(int iCnt=iStart;iCnt<=iEnd;iCnt++)
		{
			if(iCnt%2==0)
			{
				System.out.print(iCnt+" ");
			}	
		}

		System.out.println();
	}
}

class Program2
{
	public static void main(String[] args)
	{
		Scanner sObj = new Scanner(System.in);

		System.out.println("Enter Start: ");
		int iStart = sObj.nextInt();

		System.out.println("Enter End: ");
		int iEnd = sObj.nextInt();
		
		Solution obj = new Solution(iStart,iEnd);
		obj.RangeEvenDisplay();
	}
}

/*
OUTPUT:

Enter Start: 
23
Enter End: 
35
24 26 28 30 32 34 

*/
